js/pages/estimation/cca-manager.js

```diff
@@ -0,0 +1,580 @@
+					}
+				});
+
+				var comp = new Comparison({
+					target: target, 
+					comparator: comparator,
+					outcomes: outcomes,
+				});
+				this.comparisons.push(comp);
+			});
+			csXref.forEach((xref) => {
+				// Find the concept set each item
+				var selectedConceptSetList = conceptSets.filter((cs) => { return cs.id === xref.conceptSetId});
+				if (selectedConceptSetList.length == 0) {
+					console.error("Concept Set: " + xref.conceptSetId + " not found in specification.");
+				}
+				var selectedConceptSet = new ConceptSet({id: selectedConceptSetList[0].id, name: selectedConceptSetList[0].name()});
+				if (xref.targetName === constants.conceptSetCrossReference.targetComparatorOutcome.targetName) {
+					if (xref.propertyName === constants.conceptSetCrossReference.targetComparatorOutcome.propertyName.includedCovariateConcepts) {
+						this.comparisons()[xref.targetIndex].includedCovariateConceptSet(selectedConceptSet);
+					}
+					if (xref.propertyName === constants.conceptSetCrossReference.targetComparatorOutcome.propertyName.excludedCovariateConcepts) {
+						this.comparisons()[xref.targetIndex].excludedCovariateConceptSet(selectedConceptSet);
+					}
+				} else if (xref.targetName === constants.conceptSetCrossReference.negativeControlOutcomes.targetName) {
+					this.comparisons()[xref.targetIndex].negativeControlOutcomesConceptSet(selectedConceptSet);
+				} else if (xref.targetName == constants.conceptSetCrossReference.analysisCovariateSettings.targetName) {
+					var targetAnalysis = this.estimationAnalysis().estimationAnalysisSettings.analysisSpecification.cohortMethodAnalysisList()[xref.targetIndex];
+					if (xref.propertyName === constants.conceptSetCrossReference.analysisCovariateSettings.propertyName.includedCovariateConcepts) {
+						targetAnalysis.getDbCohortMethodDataArgs.covariateSettings.includedCovariateConceptSet(selectedConceptSet);
+					}
+					if (xref.propertyName === constants.conceptSetCrossReference.analysisCovariateSettings.propertyName.excludedCovariateConcepts) {
+						targetAnalysis.getDbCohortMethodDataArgs.covariateSettings.excludedCovariateConceptSet(selectedConceptSet);
+					}
+				} else if (xref.targetName == constants.conceptSetCrossReference.positiveControlCovariateSettings.targetName) {
+					var targetPcsCovariateSettings = this.estimationAnalysis().positiveControlSynthesisArgs().covariateSettings;
+					if (xref.propertyName === constants.conceptSetCrossReference.positiveControlCovariateSettings.propertyName.includedCovariateConcepts) {
+						targetPcsCovariateSettings.includedCovariateConceptSet(selectedConceptSet);
+					}
+					if (xref.propertyName === constants.conceptSetCrossReference.positiveControlCovariateSettings.propertyName.excludedCovariateConcepts) {
+						targetPcsCovariateSettings.excludedCovariateConceptSet(selectedConceptSet);
+					}
+				}
+			});
+		}
+
+		
+		init() {
```

Comment:
each component which is being used as a top-level view (in other words, goes to `router.setCurrentView`), should extend `Page` class. This class has a lifesycle method called `onPageCreated`